tests/pytests/test_language.py

```diff
@@ -0,0 +1,144 @@
+
+    # Search for "orange" using Italian
+    # It should return 1 results using invalid stemming in Italian
+    res = env.cmd('FT.SEARCH', 'idx', 'oranges', 'language', 'italian')
+    env.assertEqual(res, [1, '{word}:4', ['$', '{"word":"oranges"}']])
+
+    # drop index
+    env.cmd('FT.DROP', 'idx')
+
+def testSearchIndexLanguageField(env):
+    conn = getConnectionByEnv(env)
+    
+    # Create sample data in Italian
+    conn.execute_command('HSET', '{word}:1', 'word', 'arancia',
+                         '__lang', 'italian') # orange
+    conn.execute_command('HSET', '{word}:2', 'word', 'arance',
+                         '__lang', 'italian') # oranges
+    conn.execute_command('HSET', '{word}:3', 'word', 'mela',
+                         '__lang', 'italian') # apple
+    conn.execute_command('HSET', '{word}:4', 'word', 'mele',
+                         '__lang', 'italian') # apples
+    conn.execute_command('HSET', '{word}:5', 'word', 'orange',
+                         '__lang', 'english')
+    conn.execute_command('HSET', '{word}:6', 'word', 'oranges',
+                         '__lang', 'english')
+
+    # Create index - language Italian by default
+    env.cmd('FT.CREATE', 'idx', 'ON', 'HASH', 'PREFIX', '1', '{word}:',
+            'LANGUAGE', 'italian', 'LANGUAGE_FIELD', '__lang',
+            'SCHEMA', 'word', 'TEXT', '__lang', 'TEXT')
+
+    # Wait for index to be created
+    waitForIndex(env, 'idx')
+
+    # Search for "arancia", without passing language argument, should use 
+    # language Italian, because it is part of the index schema.
+    # It should return 2 results using stemming in Italian
+    res = env.cmd('FT.SEARCH', 'idx', 'arancia', 'SORTBY', 'word', 'DESC')
+    expected = [2, '{word}:1', ['word', 'arancia'], '{word}:2', ['word', 'arance']]
+    env.assertEqual(res, expected)
+
+    # Search for "arance", without passing language argument, should use 
+    # language Italian, because it is part of the index schema
+    # it should return 2 results using stemming in Italian
+    res = env.cmd('FT.SEARCH', 'idx', 'arance', 'SORTBY', 'word', 'DESC')
+    expected = [2, '{word}:1', ['word', 'arancia'], '{word}:2', ['word', 'arance']]
+    env.assertEqual(res, expected)
+
+    # Search for "arancia", passing language argument will override the language in the schema
+    # It should return 1 results using invalid stemming in English
+    res = env.cmd('FT.SEARCH', 'idx', 'arancia', 'SORTBY', 'word', 'DESC', 'LANGUAGE', 'english')
+    expected = [1, '{word}:1', ['word', 'arancia']]
+    env.assertEqual(res, expected)
+
+    # Search for "oranges", passing language argument will override the language in the schema
+    # It should return 2 results using stemming in English
+    res = env.cmd('FT.SEARCH', 'idx', 'oranges', 'SORTBY', 'word', 'ASC', 'LANGUAGE', 'english')
+    expected = [2, '{word}:5', ['word', 'orange'], '{word}:6', ['word', 'oranges']]
+    env.assertEqual(res, expected)
+
```

Comment:
How about adding a test searching for "orange" without a language in the query? Just to see that the default language of the index is used, and you get one result (orange..).

Reply:
I added more tests. The last one is still failing. I need to fix it.